solidity/test/token_stake/TestTokenStakeLock.js

```diff
@@ -0,0 +1,247 @@
+        stakingContract.lockStake(operator, lockPeriod, {from: operatorContract2}),
+        "Not authorized"
+      )
+    })
+
+    it("should not permit locks from disabled operator contracts", async () => {
+      await increaseTimeTo(createdAt + initializationPeriod + 1)
+      await registry.disableOperatorContract(operatorContract)
+      await expectThrowWithMessage(
+        stakingContract.lockStake(operator, lockPeriod, {from: operatorContract}),
+        "Operator contract is not approved"
+      )
+    })
+
+    it("should not permit locks from unapproved operator contracts", async () => {
+      await increaseTimeTo(createdAt + initializationPeriod + 1)
+      await expectThrowWithMessage(
+        stakingContract.lockStake(operator, lockPeriod, {from: operator}),
+        "Operator contract is not approved"
+      )
+    })
+  })
+
+  describe("single lock", async () => {
+    before(async () => {
+      operator = operator2;
+      let tx = await delegate(operator, stakingAmount)
+      await stakingContract.authorizeOperatorContract(
+        operator,
+        operatorContract,
+        { from: authorizer },
+      );
+
+      createdAt = (await web3.eth.getBlock(tx.receipt.blockNumber)).timestamp
+      await increaseTimeTo(createdAt + initializationPeriod + 1)
+      await stakingContract.lockStake(operator, lockPeriod, {from: operatorContract})
+    })
+
+    it("should only permit recover unlocked stake", async () => {
+      await undelegate(operator)
+      await expectThrowWithMessage(
+        stakingContract.recoverStake(operator),
+        "Can not recover locked stake"
+      )
+
+      await expectThrowWithMessage(
+        stakingContract.unlockStake(operator),
+        "Not authorized"
+      )
+
+      await stakingContract.unlockStake(operator, {from: operatorContract})
+      await stakingContract.recoverStake(operator)
+      // ok, no revert
+    })
+
+    it("should allow recover locked stake after lock duration has expired", async () => {
+      await undelegate(operator)
+      await expectThrowWithMessage(
+        stakingContract.recoverStake(operator),
+        "Can not recover locked stake"
+      )
+
+      await increaseTime(lockPeriod)
+      await stakingContract.recoverStake(operator, {from: operator})
+      // ok, no revert
+    })
+
+    it("should allow recover locked stake after operator contract has been disabled", async () => {
+      await undelegate(operator)
+      await expectThrowWithMessage(
+        stakingContract.recoverStake(operator),
+        "Can not recover locked stake"
+      )
+
+      // disable operator contract with panic button
+      await registry.disableOperatorContract(operatorContract)
+
+      await stakingContract.recoverStake(operator, {from: operator})
+      // ok, no revert
+    })
+
+    it("should be able to reduce the duration of existing locks", async () => {
+      await stakingContract.lockStake(operator, undelegationPeriod, {from: operatorContract})
```

Comment:
Why `undelegationPeriod` and not e.g. `duration.minutes(5)`? I think we wanted to use just some lower value and using `undelegationPeriod` suggests there is something more going on here.

Reply:
Addressed in latest